DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Helpers/DescribeContentItemHelper.cs

```diff
@@ -0,0 +1,208 @@
+                withStringBuilder.AppendLine("return nodeAndOutRelationshipsAndTheirInRelationships");
+                commandsToReturn.Add($"{command} {withStringBuilder}");
+            }
+
+            return commandsToReturn;
+        }
+
+        private async Task<IEnumerable<ContentItemRelationship>> GetRelationships(IDescribeRelationshipsContext context, List<ContentItemRelationship> currentList, IDescribeRelationshipsContext parentContext)
+        {
+            if (context.AvailableRelationships != null)
+            {
+                foreach (var child in context.AvailableRelationships)
+                {
+                    if (child != null)
+                    {
+                        context.CurrentDepth = parentContext.CurrentDepth + 1;
+                        var parentRelationship = parentContext.AvailableRelationships.FirstOrDefault(x => x.Destination.All(child.Source.Contains));
+
+                        if (parentRelationship != null && !string.IsNullOrEmpty(parentRelationship.RelationshipPathString))
+                        {
+                            var relationshipString = $"{parentRelationship.RelationshipPathString}-[r{context.CurrentDepth}:{child.Relationship}]-(d{context.CurrentDepth}:{string.Join(":", child.Destination!)})";
+                            child.RelationshipPathString = relationshipString;
+                        }
+                        else
+                        {
+                            context.CurrentDepth = 1;
+                            child.RelationshipPathString = $@"match (s:{string.Join(":", context.SourceNodeLabels)} {{{context.SourceNodeIdPropertyName}: '{context.SourceNodeId}'}})-[r{1}:{child.Relationship}]-(d{1}:{string.Join(":", child.Destination!)})";
+                        }
+                    }
+                }
+
+                currentList.AddRange(context.AvailableRelationships);
+            }
+
+            foreach (var childContext in context.ChildContexts)
+            {
+                await GetRelationships((IDescribeRelationshipsContext)childContext, currentList, context);
+            }
+
+            return currentList;
+        }
+
+        public async Task BuildRelationships(ContentItem contentItem, IDescribeRelationshipsContext context, string sourceNodeIdPropertyName, string sourceNodeId, IEnumerable<string> sourceNodeLabels)
+        {
+            var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(contentItem.ContentType);
+
+            _graphSyncHelper.ContentType = contentItem.ContentType;
+
+            var itemContext = contentItem == RootContentItem ? context : new DescribeRelationshipsContext(sourceNodeIdPropertyName, sourceNodeId, sourceNodeLabels, contentItem, _graphSyncHelper, _contentManager, _publishedContentItemVersion, context, _serviceProvider);
```

Comment:
This hard codes `_publishedContentItemVersion`.
I think it needs to use published or preview appropriately.